File: Demo/GtSdkDemo-Manual/GtSdkDemo/Controller/Advance/AliasUnbindViewController.m
//
//  AliasUnbindViewController.m
//  GtSdkDemo
//
//  Created by ak on 2020/03/20.
//  Copyright © 2019 Gexin Interactive (Beijing) Network Technology Co.,LTD. All rights reserved.
//

#import "AliasUnbindViewController.h"

@interface AliasUnbindViewController ()
@property (nonatomic, weak) IBOutlet UITextField *unbindAliasTextField;
@property (nonatomic, weak) IBOutlet UIButton *ensureUnBindAliasBtn;

@end

@implementation AliasUnbindViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"解绑别名";
    self.ensureUnBindAliasBtn.layer.cornerRadius = 6;
    self.unbindAliasTextField.attributedPlaceholder = [[NSAttributedString alloc] initWithString:self.unbindAliasTextField.placeholder attributes:@{NSForegroundColorAttributeName:[UIColor lightGrayColor]}];
}

- (IBAction)ensureUnBindAliasBtn:(id)sender {
    if (F_IsStringValue_GtEmpty(self.unbindAliasTextField.text)) {
        [Utils AlertControllerWithTitle:@"alias不能为空" andMessage:@""];
        return;
    }
    [GeTuiSdk unbindAlias:self.unbindAliasTextField.text andSequenceNum:[NSString stringWithFormat:@"%@-sign", self.unbindAliasTextField.text] andIsSelf:YES];
    [self.navigationController popViewControllerAnimated:YES];
}

@end

File: Demo/GtSdkDemo-swift/GtSdkDemo-swift/Controller/Advance/AliasUnbindViewController.swift
//
//  AliasUnbindViewController.swift
//  GtSdkDemo-swift
//
//  Created by ak on 2020/03/20.
//  Copyright © 2020 Gexin Interactive (Beijing) Network Technology Co.,LTD. All rights reserved.
//

import UIKit
import GTSDK

class AliasUnbindViewController: UIViewController {
  
  @IBOutlet weak var ensureUnBindAliasBtn: UIButton!
  @IBOutlet weak var unBindAliasTextField: UITextField!
  
  override func viewDidLoad() {
    super.viewDidLoad()
    ensureUnBindAliasBtn.layer.cornerRadius =  6
    unBindAliasTextField.attributedPlaceholder = NSAttributedString(string: unBindAliasTextField.placeholder ?? "", attributes: [.foregroundColor: UIColor.lightGray])
  }
  
  @IBAction func ensureUnBindAliasClick(_ sender: Any) {
    guard let alias = unBindAliasTextField.text, !alias.isEmpty else {
      UIAlertController.alert(title: "alias不能为空") 
      return
    }
    GeTuiSdk.unbindAlias(alias, andSequenceNum: "\(alias)-sign", andIsSelf: true)
    navigationController?.popViewController(animated: true)
  }
  
}
